src/buffer/out/textBuffer.cpp

```diff
@@ -1084,11 +1111,8 @@ void TextBuffer::SetCurrentLineRendition(const LineRendition lineRendition, cons
         // And if it's no longer single width, the right half of the row should be erased.
         if (lineRendition != LineRendition::SingleWidth)
         {
-            const auto fillChar = L' ';
             const auto fillOffset = GetLineWidth(rowIndex);
-            const auto fillLength = gsl::narrow<size_t>(GetSize().Width() - fillOffset);
-            const OutputCellIterator fillData{ fillChar, fillAttributes, fillLength };
-            row.WriteCells(fillData, fillOffset, false);
+            FillRect({ fillOffset, rowIndex, til::CoordTypeMax, rowIndex }, L" ", fillAttributes);

```

Comment:
This change isn't working either, assumedly because the `FillRect` method is constrained by the width of the line, which has now been shortened because of the altered line rendition. You might just need to move the fill above the `SetLineRendition` call.

Reply:
I just realised this is covered by your third test case and that fails for me too. I'm wondering now if maybe I've got a build issue. Does that test case definitely work for you?

Reply:
I did a rebase at some point and I likely did something wrong. I'll try to fix this soon. Thanks!